homeassistant/components/sia/config_flow.py

```diff
@@ -0,0 +1,232 @@
+        return SIAOptionsFlowHandler(config_entry)
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self._data: ConfigType = {}
+        self._options: Mapping[str, Any] = {CONF_ACCOUNTS: {}}
+
+    async def async_step_user(self, user_input: ConfigType = None):
+        """Handle the initial user step."""
+        errors: dict[str, str] | None = None
+        if user_input is not None:
+            errors = validate_input(user_input)
+        if user_input is None or errors is not None:
+            return self.async_show_form(
+                step_id="user", data_schema=HUB_SCHEMA, errors=errors
+            )
+        return await self.async_handle_data_and_route(user_input)
+
+    async def async_step_add_account(self, user_input: ConfigType = None):
+        """Handle the additional accounts steps."""
+        errors: dict[str, str] | None = None
+        if user_input is not None:
+            errors = validate_input(user_input)
+        if user_input is None or errors is not None:
+            return self.async_show_form(
+                step_id="add_account", data_schema=ACCOUNT_SCHEMA, errors=errors
+            )
+        return await self.async_handle_data_and_route(user_input)
+
+    async def async_handle_data_and_route(self, user_input: ConfigType):
+        """Handle the user_input, check if configured and route to the right next step or create entry."""
+        self._update_data(user_input)
+        if self._data and self._port_already_configured():
+            return self.async_abort(reason="already_configured")
+
+        if user_input[CONF_ADDITIONAL_ACCOUNTS]:
+            return await self.async_step_add_account()
+        return self.async_create_entry(
+            title=TITLE.format(self._data[CONF_PORT]),
+            data=self._data,
+            options=self._options,
+        )
+
+    def _update_data(self, user_input: ConfigType) -> None:
+        """Parse the user_input and store in data and options attributes.
+
+        If there is a port in the input or no data, assume it is fully new and overwrite.
+        Add the default options and overwrite the zones in options.
+        """
+        if not self._data or user_input.get(CONF_PORT):
+            self._data = {
+                CONF_PORT: user_input[CONF_PORT],
+                CONF_PROTOCOL: user_input[CONF_PROTOCOL],
+                CONF_ACCOUNTS: [],
+            }
+        account = user_input[CONF_ACCOUNT]
+        self._data[CONF_ACCOUNTS].append(
+            {
+                CONF_ACCOUNT: account,
+                CONF_ENCRYPTION_KEY: user_input.get(CONF_ENCRYPTION_KEY),
+                CONF_PING_INTERVAL: user_input[CONF_PING_INTERVAL],
+            }
+        )
+        self._options[CONF_ACCOUNTS].setdefault(account, deepcopy(DEFAULT_OPTIONS))
+        self._options[CONF_ACCOUNTS][account][CONF_ZONES] = user_input[CONF_ZONES]
+
+    def _port_already_configured(self):
```

Comment:
Can we use this instea?

https://github.com/home-assistant/core/blob/51c8b1eb0b2d0e07610bc753883ae2873fa5857e/homeassistant/config_entries.py#L1103-L1106

Reply:
I had a look, but only the Port has to be unique, and `_async_abort_entries_match` matches the full data in the setup vs only 1 field, I was using the port as unique ID, but that also didn't match, so let me know what you think.

Reply:
If we want to match against the port item we pass the port item (key and value) and it will be matched against the corresponding item in the existing entries' data.